core/src/impl/Kokkos_AnalyzePolicy.hpp

```diff
@@ -305,40 +306,36 @@ struct AnalyzePolicy<Base> {
                        is_graph_kernel, occupancy_control>;
 };
 
-template <bool Enable = false>
-struct MaybeDesiredOccupancy {};
+template <class AnalyzedPolicy>
+struct PolicyDataStorage : AnalyzedPolicy,
+                           NoUniqueAddressMemberEmulation<
+                               typename AnalyzedPolicy::occupancy_control> {
+  using occupancy_control_storage_base_t = NoUniqueAddressMemberEmulation<
+      typename AnalyzedPolicy::occupancy_control>;
 
-template <>
-struct MaybeDesiredOccupancy<true> {
-  Kokkos::Experimental::DesiredOccupancy m_occupancy;
-};
-
-template <typename... Traits>
-struct PolicyTraits
-    : public AnalyzePolicy<PolicyTraitsBase<>, Traits...>::type,
-      private MaybeDesiredOccupancy<std::is_same<
-          typename AnalyzePolicy<PolicyTraitsBase<>,
-                                 Traits...>::type::occupancy_control,
-          Kokkos::Experimental::DesiredOccupancy>::value> {
-  using traits = typename AnalyzePolicy<PolicyTraitsBase<>, Traits...>::type;
+  using occupancy_control_t = typename AnalyzedPolicy::occupancy_control;
 
   static constexpr bool experimental_contains_desired_occupancy =
-      std::is_same<typename traits::occupancy_control,
+      std::is_same<occupancy_control_t,
                    Kokkos::Experimental::DesiredOccupancy>::value;
 
-  template <bool Enable = experimental_contains_desired_occupancy>
-  std::enable_if_t<Enable> experimental_set_desired_occupancy(
-      Experimental::DesiredOccupancy desired_occupancy) {
-    this->m_occupancy = {desired_occupancy};
+  auto experimental_get_desired_occupancy() const {
```

Comment:
I actually meant to SFINAE away this setter and the getter below when `experimental_contains_desired_occupancy` is `false`